### PR TITLE
Add support for pip caching.

### PR DESCRIPTION
**Pull request by mythmon**
This isn't quite working yet. The tests are a little funky, and I need to add CLI flag support

* [x] envvar support
* [x] positive test
* [ ] negative test
* [ ] confidence in tests
* [ ] cli flag support

### CODE REVIEW

**Comment by Pike** [comment]
@mythmon, it'd be great to have this. Not great enough for me to take it, though :-) I'd offer bribe, though.

**Comment by mythmon** [comment]
This would be great to have for me too, I just haven't gotten around to fixing it up. Hopefully soon!

File: tests/__init__.py [hunk 1]
```diff
@@ -327,12 +330,53 @@ def test_non_hash_comments(self):
     def test_whitespace_stripping(self):
         """Make sure trailing whitespace is stripped from hashes."""
         reqs = self.downloaded_reqs("""
-            # sha256: trailing_space_should_be_stripped   
+            # sha256: trailing_space_should_be_stripped
```
Comment (erikrose) [inline tests/__init__.py]: I fixed this on master.

**Comment by mythmon** [comment]
I'm going to close this out. Pip 6.0 changed how caching is done. This should be redone, re-using as much of the Pip caching architecture as possible.

**Pull request closed without merging**